docs/website/contents/for-developers/Ticking.md

```diff
@@ -0,0 +1,98 @@
+(It's important to note that such a request is not doomed: the node might just need to process another block or two before the slot of interest will be within the forecast range of its selected chain's tip.)
+
+However, the Consensus Layer itself will never need to tick farther than the forecast range.
+The ledger rules might actually be able to do so, since it's mostly just a matter of the state _not_ changing.
+It could plausibly cause problems within the ledger rules if an entire epoch has no blocks, for example, so there's definitely no guarantee that the resulting ledger state would actually be useful.
+But again: the Consensus Layer will never request such a tick.
+
+The inductive argument is that all the ticking the Consensus Layer does is ultimately guarded by some forecast, unless someone has altered the node's files between executions (or if there's a bad bug somewhere else).
+
+- The mempool only ticks by a single slot, so any non-degenerate forecast range supports that.
+- The leadership check immediately aborts if it cannot forecast the `LedgerView`.
+  The mint only proceeds if the leadership succeeded.
+- The ChainSync client forecasts the `LedgerView` before it applies any header.
+- The ChainSel logic only applies blocks whose headers arrived via a ChainSync client or were minted by this node.
+  The node also processes blocks from its on-disk storage when initializing, but those blocks were either minted by this node, acquired by this node via ChainSync (and BlockFetch), written by another trusted node (eg via Mithril) between executions, or written by some other means that is not explicitly unsupported.
+
+Thus, it's reasonable for the ticking function to have a total type because none of its calls within the node can fail; it's always guarded by a forecast.
+
+## Cross-Era Ticking
+
+The only time we have ever envisioned wanting to tick a ticked ledger state --- despite our current types preventing it --- is in the Hard Fork Combinator.
+
+In particular, the HFC must tick from one era into the next era when appropriate.
+In today's code, the HFC achieves this by translating the ledger state after the final block of an era into the next era, and then ticking that ledger state across the era boundary.
+But that involves a ledger state of the next era existing in a slot before that era starts, which is confusing and has lead to some errors in the past.
+Conceptually, it'd be much clearer to tick the ledger state up to the era boundary, then translate it, and then tick the rest of the way.
+But our current `Ticked` typing disallows that kind of incremental ticking, unless the translation's type does unticking, and that also risks confusion.
+
+It's worth noting that the HFC already uses a custom interface for forecasting across eras.
+We could introduce a similar interface for ticking across eras.
+The implementation, which is lower-level than the `Ticked`-based interface, could use the less surprising tick-translate-tick scheme.
```

Comment:
This seems like an actionable item! Shall we factor it out into an issue and perhaps schedule it for the next quarter?

Reply:
The ticket for the general HFC cross-era ticking is #345, and there is an implementation in #1295, but the devil is in the details, see https://github.com/IntersectMBO/ouroboros-consensus/pull/1295#r1949044173. That PR does not switch to the tick-translate-tick scheme, but would easily enable that (though I am not sure whether that is actually *that* interesting/useful, Ledger seemed fine in guaranteeing to us that we can use the translate-then-tick scheme.)

(There also is an older implementation to enforce the tick-translate-tick scheme at the HFC level, #340, but we don't want that, see https://github.com/IntersectMBO/ouroboros-consensus/issues/339#issuecomment-1758097242.)

Reply:
I added a link to 345, thanks!